src/odemis/driver/lakeshore.py

```diff
@@ -108,11 +103,75 @@ def __init__(self, name, role, port, sensor_input='b', output_channel=2, **kwarg
         except LakeshoreError as ex:
             logging.warning("Discarding initial error status: %s", ex)
 
-        # Vigilant attributes of the controller.
-        self.temperature = FloatVA(unit=u"°C", value=self.GetSensorTemperature(), readonly=True)
-        self.targetTemperature = FloatContinuous(value=self.GetSetpoint(), unit=u"°C", range=[-273, 50], setter=self._set_targetTemperature)
-        self.heating = IntEnumerated(value=self.GetHeaterRange(), choices={0: "Off", 1: "Low", 2: "Medium", 3: "High"},
-                                     setter=self._set_heating)
+        # set supported sensor input and output channel values
+        if md == "MODEL335":
+            supported_sensor_inputs = ('A', 'B')
+            supported_output_channels = (1, 2)
+        elif md == "MODEL350":
+            supported_sensor_inputs = ('A', 'B', 'C', 'D')
+            supported_output_channels = (1, 2, 3, 4)
+        else:
+            raise IOError("Incompatible device model detected. Supported models: MODEL350 and MODEL335")
+
+        # Check if there's only one sensor input and the input is defined as string
+        if isinstance(sensor_input, str):
+            sensor_input = {"": sensor_input}
+        # Check if there's one or multiple sensor input(s) and defined as dictionary
```

Comment:
this comment does not completely match the line below. I would say something like: `Make sure the input is either a single string or a dictionary.`. It's also possible to remove this comment as it does not add information you do not get from reading the code.